File: LC209.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i = 0; l = float('inf')
        n = len(nums)
        s = 0

        for j in range(n):
            s += nums[j]

            while s >= target:
                s -= nums[i]
                l = min(l, j - i + 1)
                i += 1
            
        return 0 if l == float('inf') else l

File: LC209.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n = nums.size(), i = 0, len = INT_MAX, sum = 0;

        for(int j = 0; j < n; j++){

            sum += nums[j];

            while(sum >= target){
                len = min(len, j - i + 1);
                sum -= nums[i++];
            }
        }

        return len == INT_MAX? 0 : len;
    }
};
